cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -475,30 +480,42 @@ private void storeActiveConnection(final JSONObject config) {
   // Notifications
 
   /* Displays a persistent notification that sets a pending intent to open the app. */
-  private void displayPersistentNotification(final JSONObject serverConfig) {
+  private void displayPersistentNotification(
+      final JSONObject serverConfig, OutlinePlugin.ConnectionStatus status) {
     try {
-      Intent launchIntent = new Intent(this, getPackageMainActivityClass());
-      PendingIntent mainActivityIntent =
-          PendingIntent.getActivity(this, 0, launchIntent, PendingIntent.FLAG_UPDATE_CURRENT);
-      Notification.Builder builder =
-          new Notification.Builder(this)
-              .setContentTitle(getServerName(serverConfig))
-              .setContentText("Connected")
-              .setSmallIcon(getResourceIdForDrawable("small_icon"))
-              .setColor(NOTIFICATION_COLOR)
-              .setVisibility(Notification.VISIBILITY_SECRET) // Don't display in lock screen
-              .setContentIntent(mainActivityIntent)
-              .setShowWhen(true)
-              .setUsesChronometer(true);
-      startForeground(NOTIFICATION_SERVICE_ID, builder.getNotification());
+      if (notificationBuilder == null) {
```

Comment:
There's quite a bit going on here. Is there any way for this function to query the "current server"? If so, you wouldn't have to "cache" the builder like this.

Reply:
There's a few options: I can keep an instance variable for the connection configuration (or server name), or retrieve it from the connection store. I decided to cache the notification builder because we don't need to reconstruct it every time the connectivity changes.

Reply:
Another reason for caching the builder is to update the existing notification, instead of creating a new one when connectivity changes. This is noticeable for the user because the timer resets when a new builder is used.

Reply:
OK, worth mentioning in a comment.

Reply:
Done in c93d350.